### ISSUE TITLE
Erro na compilação

### ISSUE THREAD

**Issue by farias2020** [alberteije/NFC-e-Desktop-Delphi#1]
Olá Albert Eije. Tudo bem?

Instalei os componentes conforme orientação da video aula; Contudo não consigo compilar, pois surgem os seguintes erros abaixo que não consigo solucionar.

[dcc32 Error] Biblioteca.pas(539): E2003 Undeclared identifier: 'DecimalSeparator'
[dcc32 Error] Biblioteca.pas(539): E2250 There is no overloaded version of 'Pos' that can be called with these arguments
[dcc32 Fatal Error] T2TiERPNFe.dpr(6): F2063 Could not compile used unit 'Biblioteca.pas'

Estou usando o Delphi XE7; Será que você poderia me dar um apoio?

Agradeço!

**Comment by alberteije** [comment]
Por conta de sua versão do Delphi, deverá realizar algumas adaptações. Se buscar no google, encontrará algumas dicas.

Exemplo:
https://stackoverflow.com/questions/31559128/errors-undeclared-identifier-decimalseparator-and-no-overloaded-version-of-s

Para o decimal separator vai utilizar agora dessa forma: FormatSettings.DecimalSeparator

Provavelmente esse problema está levando ao outro do POS e quando arrumar o primeiro o segundo deve desaparecer.

**Comment by farias2020** [comment]
> Por conta de sua versão do Delphi, deverá realizar algumas adaptações. Se buscar no google, encontrará algumas dicas.
> 
> Exemplo: https://stackoverflow.com/questions/31559128/errors-undeclared-identifier-decimalseparator-and-no-overloaded-version-of-s
> 
> Para o decimal separator vai utilizar agora dessa forma: FormatSettings.DecimalSeparator
> 
> Provavelmente esse problema está levando ao outro do POS e quando arrumar o primeiro o segundo deve desaparecer.

Albert Eije. Primeiramente quero te agradecer por ter respondido. 
Eu gostaria de uma sugestão sua. Eu já corrigir alguns erros e por sinal avancei bastante, mesmo sendo novo com a ferramenta. As telas já abrem sem erros. 

Mas estou com esses erros abaixo, e mesmo seguindo orientações do stackoverflow e outros fóruns, não avancei.
Você poderia meu ajudar neste? Te pergunto, seria melhor instalar a versão XE 5 ou você sugere ficar corrigindo nesta que estou?

[dcc32 Error] VO.pas(310): E2003 Undeclared identifier: 'TJSONValue'
[dcc32 Fatal Error] Biblioteca.pas(43): F2063 Could not compile used unit 'VO.pas' 

Obrigado pelo apoio.

**Comment by alberteije** [comment]
Sim, esses problemas ocorrem por conta da versão do Delphi. Essa TJsonValue fazia parte de uma unit e agora faz parte de outra (ou algo assim), daí você deve buscar no google algo parecido com isso: TJSONValue Delphi XE5 XE7. Ou então crie um novo projeto e utilize os componentes que trabalham com JSON e observe quais Units que aparecem na seção Uses. Daí você compara com o que está nesse código do XE5 e faz a adaptação. Não precisa voltar para o XE5, só fazer essas adaptações que vai dá certo e você já vai saber o que fazer quando ocorrerem problemas similares ao migrar para uma versão mais recente do Delphi.
